modules/objdetect/src/aruco/fractal_detector.cpp

```diff
@@ -187,244 +185,40 @@ void FractalArucoMarker::push_back(const cv::Point2f& p)
     impl->data.push_back(p);
 }
 
-
-// Implementation of FractalMarkerSet methods
-FractalMarkerSet::FractalMarkerSet(const std::string& config) {
+FractalMarkerDictionary::FractalMarkerDictionary()
+    : impl(makePtr<FractalMarkerDictionaryImpl>())
+{
+    impl->mInfoType = -1;
+    impl->idExternal = -1;
+}
+// Implementation of FractalMarkerDictionary methods
+FractalMarkerDictionary::FractalMarkerDictionary(const std::string& config)
+    : impl(makePtr<FractalMarkerDictionaryImpl>())
+{
```

Comment:
Could you add some description of the binary data layout for predefined markers or at least link to a paper how they are generated. The main purpose of the dedicated `FractalArucoDictionary` is a way to create and use own dictionaries. The data layout and generation tools should be more or less obvious from the documentation.